File: MailLibTest/DispositionBuilderTest.cs
using System;
using NUnit.Framework;
using PeterO.Mail;

namespace MailLibTest {
  [TestFixture]
  public class DispositionBuilderTest {
    [Test]
    public void TestConstructor() {
      string stringNull = null;
      ContentDisposition dispNull = null;
      try {
        Assert.AreEqual(null, new ContentDisposition.Builder(stringNull));
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        Assert.AreEqual(null, new ContentDisposition.Builder(dispNull));
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        Assert.AreEqual(null, new ContentDisposition.Builder(String.Empty));
        Assert.Fail("Should have failed");
      } catch (ArgumentException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
    }
    [Test]
    public void TestDispositionType() {
      var db = new ContentDisposition.Builder();
      db.SetDispositionType("inline");
      Assert.AreEqual("inline", db.DispositionType);
    }
    [Test]
    public void TestRemoveParameter() {
      try {
        new ContentDisposition.Builder().RemoveParameter(null);
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
    }
    [Test]
    public void TestSetDispositionType() {
      try {
        new ContentDisposition.Builder().SetDispositionType(null);
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetDispositionType(String.Empty);
        Assert.Fail("Should have failed");
      } catch (ArgumentException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
    }
    [Test]
    public void TestSetParameter() {
      var db = new ContentDisposition.Builder().SetParameter("a", "b");
      {
        string stringTemp = db.ToDisposition().GetParameter("a");
        Assert.AreEqual(
          "b",
          stringTemp);
      }
      db.SetParameter("a", String.Empty);
      Assert.AreEqual(String.Empty, db.ToDisposition().GetParameter("a"));
      try {
        new ContentDisposition.Builder().SetParameter(null, null);
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter(null, "test");
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter(null, String.Empty);
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test", null);
        Assert.Fail("Should have failed");
      } catch (ArgumentNullException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test", String.Empty);
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter(String.Empty, "value");
        Assert.Fail("Should have failed");
      } catch (ArgumentException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test\u00e0", "value");
        Assert.Fail("Should have failed");
      } catch (ArgumentException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
    }
    [Test]
    public void TestToDisposition() {
      // not implemented yet
    }
    [Test]
    public void TestToString() {
      var disp = new ContentDisposition.Builder();
      disp.SetDispositionType("attachment");
      disp.SetParameter("a", "b");
      {
        string stringTemp = disp.ToString();
        Assert.AreEqual(
          "attachment;a=b",
          stringTemp);
      }
    }
  }
}

File: src/test/java/com/upokecenter/test/DispositionBuilderTest.java
package com.upokecenter.test; import com.upokecenter.util.*;

import org.junit.Assert;
import org.junit.Test;
import com.upokecenter.mail.*;

  public class DispositionBuilderTest {
    @Test
    public void TestConstructor() {
      String stringNull = null;
      ContentDisposition dispNull = null;
      try {
        Assert.assertEquals(null, new ContentDisposition.Builder(stringNull));
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        Assert.assertEquals(null, new ContentDisposition.Builder(dispNull));
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        Assert.assertEquals(null, new ContentDisposition.Builder(""));
        Assert.fail("Should have failed");
      } catch (IllegalArgumentException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
    }
    @Test
    public void TestDispositionType() {
      ContentDisposition.Builder db = new ContentDisposition.Builder();
      db.SetDispositionType("inline");
      Assert.assertEquals("inline", db.getDispositionType());
    }
    @Test
    public void TestRemoveParameter() {
      try {
        new ContentDisposition.Builder().RemoveParameter(null);
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
    }
    @Test
    public void TestSetDispositionType() {
      try {
        new ContentDisposition.Builder().SetDispositionType(null);
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetDispositionType("");
        Assert.fail("Should have failed");
      } catch (IllegalArgumentException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
    }
    @Test
    public void TestSetParameter() {
      ContentDisposition.Builder db = new ContentDisposition.Builder().SetParameter("a", "b");
      {
        String stringTemp = db.ToDisposition().GetParameter("a");
        Assert.assertEquals(
          "b",
          stringTemp);
      }
      db.SetParameter("a", "");
      Assert.assertEquals("", db.ToDisposition().GetParameter("a"));
      try {
        new ContentDisposition.Builder().SetParameter(null, null);
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter(null, "test");
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter(null, "");
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test", null);
        Assert.fail("Should have failed");
      } catch (NullPointerException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test", "");
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("", "value");
        Assert.fail("Should have failed");
      } catch (IllegalArgumentException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
      try {
        new ContentDisposition.Builder().SetParameter("test\u00e0", "value");
        Assert.fail("Should have failed");
      } catch (IllegalArgumentException ex) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.fail(ex.toString());
        throw new IllegalStateException("", ex);
      }
    }
    @Test
    public void TestToDisposition() {
      // not implemented yet
    }
    @Test
    public void TestToString() {
      ContentDisposition.Builder disp = new ContentDisposition.Builder();
      disp.SetDispositionType("attachment");
      disp.SetParameter("a", "b");
      {
        String stringTemp = disp.toString();
        Assert.assertEquals(
          "attachment;a=b",
          stringTemp);
      }
    }
  }
